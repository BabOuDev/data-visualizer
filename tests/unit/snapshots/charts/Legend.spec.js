import {shallowMount} from '@vue/test-utils';
import Legend from '@/components/charts/Legend.vue';

describe('Legend.vue', () => {
  it('should renders correctly', () => {
    const component = shallowMount(Legend, {
      props: {
        keys: ['A', 'B', 'C'],
        colors: ['#f00', '#0f0', '#00f'],
      },
    });
    expect(component.html()).toMatch('A');
    expect(component.html()).toMatch('B');
    expect(component.html()).toMatch('C');
    expect(component.html()).toMatchSnapshot();
  });
});
