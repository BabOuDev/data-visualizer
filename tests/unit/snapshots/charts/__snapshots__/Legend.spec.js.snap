// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend.vue should renders correctly 1`] = `
<div class="legend-container">
  <div class="key">
    <div class="square" style="background-color: rgb(255, 0, 0);"></div><label class="label">A</label>
  </div>
  <div class="key">
    <div class="square" style="background-color: rgb(0, 255, 0);"></div><label class="label">B</label>
  </div>
  <div class="key">
    <div class="square" style="background-color: rgb(0, 0, 255);"></div><label class="label">C</label>
  </div>
</div>
`;
